Firestore/core/src/firebase/firestore/local/serializer.cc

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/local/serializer.h"
+
+#include <cstdlib>
+#include <utility>
+
+#include "Firestore/Protos/nanopb/firestore/local/maybe_document.nanopb.h"
+#include "Firestore/Protos/nanopb/google/firestore/v1beta1/document.nanopb.h"
+#include "Firestore/core/src/firebase/firestore/model/field_value.h"
+#include "Firestore/core/src/firebase/firestore/model/no_document.h"
+#include "Firestore/core/src/firebase/firestore/nanopb/tag.h"
+#include "Firestore/core/src/firebase/firestore/util/hard_assert.h"
+
+namespace firebase {
+namespace firestore {
+namespace local {
+
+using firebase::firestore::model::ObjectValue;
```

Comment:
super-mega-ultra-nit: elsewhere we've put the using statements before the namespace opens.

Reply:
It's an attempt to follow the advice here:
http://go/totw/119#scope-of-the-alias (https://abseil.io/tips/119)

I'd propose fixing the other instances, but the likelihood of this actually biting us seems... remote.

Reply:
That article also argues for fully qualifying to prevent any ambiguity.

Preventing collisions in the global namespace is a worthy goal. Let's clean this up separately.